test/pipelinefinally_test.go

```diff
@@ -55,25 +58,69 @@ func TestPipelineLevelFinally_OneDAGTaskFailed_Failure(t *testing.T) {
 		t.Fatalf("Failed to create dag Task: %s", err)
 	}
 
-	finalTask := getSuccessTask(t, namespace)
-	if _, err := c.TaskClient.Create(ctx, finalTask, metav1.CreateOptions{}); err != nil {
+	successTask := getSuccessTask(t, namespace)
+	if _, err := c.TaskClient.Create(ctx, successTask, metav1.CreateOptions{}); err != nil {
 		t.Fatalf("Failed to create final Task: %s", err)
 	}
 
-	pipeline := getPipeline(t,
-		namespace,
-		map[string]string{
-			"dagtask1": task.Name,
-			"dagtask2": delayedTask.Name,
-			"dagtask3": finalTask.Name,
+	taskConsumingResult := getSuccessTaskConsumingResults(t, namespace)
+	if _, err := c.TaskClient.Create(ctx, taskConsumingResult, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Failed to create Task consuming task results: %s", err)
+	}
+
+	we := v1beta1.WhenExpressions{{
+		Input:    "foo",
+		Operator: "notin",
+		Values:   []string{"foo"},
+	}}
+
+	dag := map[string]pipelineTask{
+		"dagtask1": {
+			TaskName: task.Name,
+		},
+		"dagtask2": {
+			TaskName: delayedTask.Name,
+		},
+		"dagtask3": {
+			TaskName:  successTask.Name,
+			Condition: cond.Name,
+		},
+		"dagtask4": {
+			TaskName: successTask.Name,
+			When:     we,
+		},
+	}
+
+	f := map[string]pipelineTask{
+		"finaltask1": {
+			TaskName: successTask.Name,
```

Comment:
would it make sense to have a final task also successfully consuming a result, for completeness?

Reply:
haha I had added it and dropped it, sure will add it back.

Reply:
done